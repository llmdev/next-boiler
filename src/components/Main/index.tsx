import * as S from './styles'

const Main = () => (
  <S.Wrapper>
    <h1>react avancado</h1>
  </S.Wrapper>
)

export default Main
