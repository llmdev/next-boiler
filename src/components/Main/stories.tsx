import Main from './index'

export default {
  title: 'Main',
  component: Main
}

export const Basic = () => <Main />
