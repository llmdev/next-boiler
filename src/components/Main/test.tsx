import { render, screen } from '@testing-library/react'

import Main from '.'

describe('Main Component', () => {
  it('should render the header', () => {
    const { container } = render(<Main></Main>)

    expect(
      screen.getByRole('heading', { name: /react avancado/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('shoudl render the corlors correct', () => {
    const { container } = render(<Main />)
    expect(container.firstChild).toHaveStyle({ 'background-color': '#c0c' })
  })
})
