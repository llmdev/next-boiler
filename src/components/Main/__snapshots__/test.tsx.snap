// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component should render the header 1`] = `
.c0 {
  background-color: #c0c;
}

<main
  class="c0"
>
  <h1>
    react avancado
  </h1>
</main>
`;
